proxy/coco/src/state.rs

```diff
@@ -1,799 +1,878 @@
 //! Utility to work with the peer api of librad.
 
-use std::{convert::TryFrom as _, net::SocketAddr, path::PathBuf, sync::Arc, time::Duration};
+use std::{convert::TryFrom as _, net::SocketAddr, path::PathBuf};
```

Comment:
This file easily has the worst diff to review. We removed the `State` struct and made all the functions top-level. They now take `&net::peer::Peer` to interact with the underlying `Storage`. The changes were mechanical and the functions should be on par with what was there.